Dockerfile

```
@@ -12,19 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Build the manager binary
 FROM registry.svc.ci.openshift.org/openshift/release:golang-1.13 as builder
-
-# Copy in the go src
 WORKDIR /go/src/sigs.k8s.io/cluster-api-provider-azure
-COPY pkg/    pkg/
-COPY cmd/    cmd/
-COPY vendor/ vendor/
-
-# Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o ./machine-controller-manager sigs.k8s.io/cluster-api-provider-azure/cmd/manager
+COPY . .
+# VERSION env gets set in the openshift/release image and refers to the golang version, which interfers with our own
+RUN unset VERSION \
+  && GOPROXY=off NO_DOCKER=1 make build
 
-# Copy the controller-manager into a thin image
 FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base
-WORKDIR /
-COPY --from=builder /go/src/sigs.k8s.io/cluster-api-provider-azure/machine-controller-manager .
+COPY --from=builder /go/src/sigs.k8s.io/cluster-api-provider-azure/machine-controller-manager /
```

This one needs a bin now based on the makefile
```suggestion
COPY --from=builder /go/src/sigs.k8s.io/cluster-api-provider-azure/bin/machine-controller-manager /
```

---

👍